packages/bun-usockets/src/loop.c

```diff
@@ -397,6 +397,36 @@ void us_internal_dispatch_ready_poll(struct us_poll_t *p, int error, int events)
             }
             break;
         }
+        case POLL_TYPE_UDP: {
+            struct us_udp_socket_t *u = (struct us_udp_socket_t *) p;
+            // do cleanup
+            if (u->closed) {
+                bsd_close_socket(us_poll_fd(p));
+                us_poll_stop(p, u->loop);
+                free(u->receive_buf);
+                us_poll_free(p, u->loop);
+                break;
+            }
+
+            if (events & LIBUS_SOCKET_WRITABLE && !error) {
+                u->on_drain(u);
+                if (u->closed) {
+                    break;
+                }
+                // TODO handle socket close in callback
+                us_poll_change(&u->p, u->loop, us_poll_events(&u->p) & LIBUS_SOCKET_READABLE);
```

Comment:
why do we call us_poll_change? readable/writable for sockets shouldn't be one shot so we shouldn't need to mark it as readable/writable each time